src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs

```diff
@@ -94,7 +95,11 @@ protected override void OnApplyTemplate()
 
 		private void OnSizeChange()
 		{
-			SetProgressBarIndicatorWidth();
+			var measures = (m_layoutRoot?.ActualWidth, m_determinateProgressBarIndicator?.ActualWidth);
+			if (m_previousMeasuredWidths != measures)
+			{
+				SetProgressBarIndicatorWidth();
+			}
 			UpdateWidthBasedTemplateSettings();
 		}
```

Comment:
Is this a workaround for a different issue?

The original code doesn't have these checks:

https://github.com/microsoft/microsoft-ui-xaml/blob/4cbade6959cda4fca42a299fb540adc2f99c1e8a/dev/ProgressBar/ProgressBar.cpp#L50-L54

Reply:
I meant the root cause for the issue might be something else and this is a workaround. It's unclear why it works on other platforms and not Android.

Reply:
The suggestion from @Xiaoy312 (https://github.com/unoplatform/uno/pull/12397#discussion_r1204305898) matches my expectation. This feels more like a workaround.

Reply:
change made